src/Directory.Build.targets

```diff
@@ -35,4 +35,83 @@
     <None Remove="$(NUGET_PACKAGES)Microsoft.Diagnostics.Tracing.TraceEvent\*\build\native\**" />
   </ItemGroup>
 
+  <Target Name="CalculateProjectRuntimeConfigTestFilePath">
+    <PropertyGroup>
+      <ProjectRuntimeConfigTestFileName Condition="'$(ProjectRuntimeConfigTestFileName)' == ''">$(AssemblyName).runtimeconfig.test.json</ProjectRuntimeConfigTestFileName>
+      <ProjectRuntimeConfigTestFilePath Condition="'$(ProjectRuntimeConfigTestFilePath)' == ''">$(TargetDir)$(ProjectRuntimeConfigTestFileName)</ProjectRuntimeConfigTestFilePath>
+    </PropertyGroup>
+  </Target>
+
+  <!--
+    Typically, the SDK will generate a runtimeconfig.json file with frameworks that it natively has included
+    with itself. This does not allow testing with newer runtime versions brought in through dependency updates.
+    The unit test apps can be rolled forward using RuntimeFrameworkVersion at build time, however we do not want
+    to do this with any binaries that are shipped e.g. dotnet-monitor. Additionally the runtime-fx argument to
+    dotnet does not work for any application that has more than one framework, such as ASP.NET apps.
+
+    The general purpose solution is to generate a custom runtimeconfig[.test].json file that instructs the
+    runtime which framework versions to use. The framework versions in this file are the ones provided by the
+    dependency flow system.
+  -->
+  <Target Name="GenerateTestRuntimeConfigurationFiles"
```

Comment:
I think we covered this, but I want to make sure that it's still possible to run unit tests via the unit test explorer

Reply:
Yes, that is correct. The runtimeconfig.test.json file is used only if the dotnet host was restored to `<repoRoot>/.dotnet/`. This means that if you've restored the dotnet host and then sync your repo to get newer runtime versions, you will have to restore those runtimes again before building in Visual Studio. If this becomes a problem, we can either scope it down so that the runtimeconfig.test.json file is only used outside of Visual Studio or we can dynamically restore the missing runtimes.